### PR TITLE
The spec.

### PR DESCRIPTION
**Pull request by TheReturnOfJan**
Resolves issue #1

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+The word _machine_ is used to refer to any deterministic state machine that implements a specified [interface](./src/Machine.template.sol).
+
+Each _machine_ is defined by 3 types: _State_, _Seed_ and _Image_, and 6 functions: _create_, _project_, _isTerminal_, _next_, _stateHash_ and _imageHash_.
+
+_State_ describes the state of the machine at a particular moment of execution.
+
+The function _next_ performs the smallest possible step of computation, mapping the input _State_ into an output _State_, **or failing** (all of the situations in which it fails will be described later).
+
+The function _isTerminal_ determines if a given _State_ is terminal, meaning that if _isTerminal(s) == true_, where _s_ is of type _State_, then _next(s)_ must either fail or return _s_ unchanged.
```
Comment (johannbarbie) [inline README.md]: > _next(s)_ must either fail or return _s_ unchanged.

is an "unchanged _s_" really possible? if next is supposed to "perform the smallest possible step of computation", then state must have changed, hence same state can not be returned.

Comment (TheReturnOfJan) [reply README.md]: I guess it depends on whether you consider doing nothing, or [no-op](https://en.wikipedia.org/wiki/NOP_(code)) to be "a step of computation".  

But the point I was trying to get across is that you define a terminal state by either saying "a terminal state is one from which no further computation can be done", or "a terminal state is a [fixpoint](https://en.wikipedia.org/wiki/Fixed_point_(mathematics)) with respect to the state-transition function".  

And actually, I think I will have to commit to one of these in future, as my implementation efforts continue.

**Comment by johannbarbie** [comment]
reviewed court and Oracle, had the following issues understanding them:
- not clear in which order the external functions will be called. actor description would be nice to have.
- Oracle "question" and "answer" struct names are very generic, don't convey what the Oracle is about.
- owner relationship between Court and Oracle expected, but not reflected in code.
- how does the setup avoid storing VM payload on chain, but ensures data availability? not obvious in code.
- no description about Oracle or Court in readme

**Comment by TheReturnOfJan** [comment]
Thank you @johannbarbie. These issues will be addressed along with the explanatory gif: https://github.com/leapdao/EthereumMachineOracle/issues/3#issuecomment-592928355. I will consider this issue closed.

**Pull request closed without merging**